A Back Button for AddGamesWindow

danieljohnson2 opened the pull request:
Or, "InstallerWindow is not enough!"

This PR is a facelift for the AddGamesWindow using the NavigationStack from InstallerWindow. It now looks like this:
![image](https://user-images.githubusercontent.com/6507403/214961037-6399ba60-91f9-484b-88c1-2787a6540031.png)There's a back button, a cancel button, a border around the list, and now the one choice that directly opens another window gets a distinct icon. All others now go to a page, with further explanation on it.

The search page looks like this:
![image](https://user-images.githubusercontent.com/6507403/211691165-5689c6f6-3b1b-4764-b627-bc3e642fe16d.png)
More buttons, and explanatory text. I think it worth saying that the installer window is coming.

The 'install from setup' page looks like this:
![image](https://user-images.githubusercontent.com/6507403/214961196-dbabff79-208a-447e-9ce6-99fd10c39c69.png)This now features controls to specify the identifier, and to opt into a 32-bit Wine prefix (only on 64-bit systems). These settings are problematic to change after installation- you can't change what the Wine prefix is without rebuilding it, and the default directory name is the identifier. Resolves #4708. Settings you can conveniently alter after installation not included.

Also, the 'Continue' button has turned blue, says 'Install' and moved down to the lower right corner. And there's text to explain what this does.

When you click 'Scan folder', you get a new page:
![Screenshot from 2023-01-13 19-23-44](https://user-images.githubusercontent.com/6507403/212441006-acc6b038-071d-41c9-87d9-435cda65f4f6.png)
You can key in a path, or click browse for the file selector. I think the explanatory text is badly needed here, as I found it hard to understand this. Lets hope I figured it out correctly!

The 'imported games' page looks like this:
![image](https://user-images.githubusercontent.com/6507403/211676107-d9cfbd0c-4890-41fe-92ff-162fa1bd066c.png)
Mostly cosmetic changes, but there's a 'Close' button now. This is to suggest that you're done now, though you can still go back too.

The 'Install from Script' option gets this page:
![image](https://user-images.githubusercontent.com/6507403/212440946-ca34e167-e94a-44b4-9b68-a67e8416b5a5.png)
This tries to explain what a Lutris script is and where you could get one, and it explains that this is not a headless install; it just runs through the installer window.

And the dialog getrs a new "Import a ROM" option; here, which leads here:
![image](https://user-images.githubusercontent.com/6507403/214961466-3baf4614-0ec2-4190-9c2b-4a7a7e3cc5a4.png)
This then leads to the dialog you can also get via drag & drop; but having this here adds a bit of discoverability.

strycore commented:
That's really nice to have!
Some few notes before I can merge this:

I wouldn't put "not recommended" next to the 32bit prefix checkbox. I'm considering having a drop-down with an OS selection with:
- Windows 10 - 64 bit (Default)
- Windows 7 - 64bit
- Windows 7 - 32bit
- Windows XP - 32bit
- Windows 98 - 32bit

For a lot of games requiring a 32bit prefix, it's also required to change the Windows version.


Installer scripts are mostly written in YAML by the users, not JSON as it is indicated in the dialog. JSON is only the transport format used by the API.

danieljohnson2 commented:
Sure thing, @strycore, I've tweaked the text here and there as suggested: no more "(not recommended)" and "YAML" for "JSON". I've updated the screenshots above, and added screenies for the "Import a ROM" option.

The text wants further revisions, I think. I'm equivocating on how much credit to give TOSEC for the ROM thing; I'm not that solid on how the new Import a ROM feature really works.

I've not added a menu of Windows versions though- I do not see how to do it. There does not seem to be an installer option for this I could use (like "arch" for the 32-bit thing), and no game configuration option for it. I could build of this, I expect, but I don't think it belongs in this PR, which is already fairly large. ~~One~~five things at a time, yes?

strycore commented:
That's cool, the drop-drop thing can be added later. Basically, what it would do is set the prefix to the wanted arch and introduce a winetricks task to set the correct version in the automated installer.

For TOSEC, we do have data that comes from other databases such as No-Intro and Redump. I'll reword the message if need be.

danieljohnson2 commented:
Feel free to correct my ramblings. I feel explanation is wanted here, but I'm explaining what I don't fully understand. Story of my life really.

I had figured I could do the windows-version setting the same way I did the DPI- by setting a registry value inside the prefix, with no winetricks dependency.

In any case, thanks for the mergitude @strycore!

realnc commented:
This has disappeared apparently? In 0.5.13, I don't get 32-bit choices anymore. Both Windows 10 and Windows 7 in the drop-down list are only 64-bit. When I try to install a game and then change to 32-bit later, it doesn't launch and I get a Wine error telling me that the prefix is 64-bit.

strycore commented:
Changing the architecture of an existing prefix isn't possible so you'll want to have the arch set to win32 from the start. I wonder why you're not seeing Win XP and Win 98 in the list. Have you tried with the latest code on git?

realnc commented:
> Changing the architecture of an existing prefix isn't possible so you'll want to have the arch set to win32 from the start. I wonder why you're not seeing Win XP and Win 98 in the list. Have you tried with the latest code on git?

XP 32-bit is there. But not 7 or 10.

danieljohnson2 commented:
Well, I think the idea was that Windows 7 32 bit was very rare, and not a likely choice. It's strange to need that combination.

You can change the version after installing, though. Install using the XP 32-bit setting, and then afterwards use the "Wine configuration" command- in the Wine menu in the action bar at the bottom of the main Lutris window.

With that command you can change the Windows version, and get your Win7 32-bit prefix.

realnc commented:
Ah, OK. Thank you.

Just in case you're wondering, using 32-bit W10 is useful for very old 32-bit games (9x/XP) that happened to get a DX11 renderer by modding. Kinda like running games with dgVoodoo2. The renderer does require a modern Windows version, but the rest of the game code might still be from the stone age and thus behave better with a 32-bit prefix.

strycore commented:
Do you have one single example of such a game?
Also, special cases are still best handled with install scripts on the website.  
I would like to reach a point where install scripts *only* deal with special cases and the rest is automated.
